crates/runtime/src/http/routes.rs

```diff
@@ -335,3 +345,33 @@ async fn check_shutdown(
 
     next.run(req).await
 }
+
+#[cfg(feature = "databricks")]
+async fn databticks_u2m_middleware(
```

Comment:
Instead of defining a static function here, we should have a trait for token provider middleware we want to run and the databricks middleware adds itself to the middleware registry. The route creation would then add all token provider middlewares from the registry.

Reply:
I will take a look at this.